lib/indexer.rb

```diff
@@ -98,18 +101,25 @@ def find_cdxj_files
   end
 
   def write_cdxj(warc_file)
-    cdxj_path = cdxj_path(warc_file)
-    cdxj_path.parent.mkpath unless cdxj_path.parent.directory?
-
-    cmd = ['cdxj-indexer', '--post-append', '--output', cdxj_path.to_s, warc_file.to_s]
-    @log.info("generating cdxj #{cdxj_path} for #{warc_file}")
-    @log.error("cdxj error while parsing #{warc}") unless system(*cmd)
+    cdxj = cdxj_path(warc_file)
+    cdxj.parent.mkpath unless cdxj.parent.directory?
+    cdxj_temp = Pathname.new(cdxj.to_s + '.tmp')
+
+    cmd = ['cdxj-indexer', '--dir-root', @warcs_dir.to_s, '--post-append', '--output', cdxj_temp.to_s, warc_file.to_s]
```

Comment:
This is the significant change, that ensures that the CDXJ files contain a path that is relative to `--warcs-dir` in our case usually `/web-archiving-stacks/data/collections`.